tty/uart16550/uart16550.c

```diff
@@ -242,14 +242,16 @@ static void poolthr(void *arg)
 
 static int _uart_init(uart_t *uart, unsigned int uartn, unsigned int speed)
 {
-	unsigned int divisor = 115200 / speed;
+	unsigned int divisor;
 	libtty_callbacks_t callbacks;
 
-	int err = uarthw_init(uartn, uart->hwctx, sizeof(uart->hwctx));
+	int err = uarthw_init(uartn, uart->hwctx, sizeof(uart->hwctx), &divisor);
```

Comment:
It seems that variable divisor has two faces. Value from `uarthw_init` is a clock, not a divisor ;) It is just a little misleading.

Reply:
I later modify this variable to get divisor. Yeah, maybe 2 variables would be most readable, but I don't find this very problematic. You're free to change this, if you wish